src/org/labkey/targetedms/outliers/OutlierGenerator.java

```diff
@@ -344,6 +374,24 @@ private void retainIfPreferred(Map<String, QCMetricConfiguration> preferredConfi
         return precursors;
     }
 
+    private boolean isExcludedPrecursorPeptide(Collection<ExcludedPrecursor> excludedPrecursors, String modifiedSeq, int charge, double mz)
+    {
+        return excludedPrecursors.stream().filter(ep -> Objects.equals(ep.getModifiedSequence(), modifiedSeq) &&
```

Comment:
You should be able to use anyMatch() here instead of filter() which would be a tiny bit more efficient. Since the DB doesn't have a constraint to prevent duplicates anymore, this would also handle the case where someone managed to double-insert an exclusion

Reply:
Good point! Now done